tests/devsim_layer_tests.py

```diff
@@ -0,0 +1,181 @@
+        os.system("{} --portability > {}".format(vulkaninfo_path, self.output_filepath))
+
+
+class PortabilityExtensionPresentTest(BasicTest):
+    def check(self):
+        with open(self.output_filepath) as output_file:
+            test_results = json.load(output_file)
+            if "ArrayOfVkExtensionProperties" in test_results:
+                for i in test_results["ArrayOfVkExtensionProperties"]:
+                    if i["extensionName"] == "VK_KHR_portability_subset":
+                        return True
+            return False
+
+
+class TestEnvironmentSettings:
+    def __init__(self, instance_layers="VK_LAYER_LUNARG_device_simulation", debug_enable=0, exit_on_error=0, filename="", portability=0, extension_list=0):
+        self.settings = {}
+        self.settings["VK_INSTANCE_LAYERS"] = instance_layers
+        self.settings["VK_DEVSIM_DEBUG_ENABLE"] = debug_enable
+        self.settings["VK_DEVSIM_EXIT_ON_ERROR"] = exit_on_error
+        self.settings["VK_DEVSIM_FILENAME"] = filename
+        self.settings["VK_DEVSIM_EMULATE_PORTABILITY_SUBSET_EXTENSION"] = portability
+        self.settings["VK_DEVSIM_MODIFY_EXTENSION_LIST"] = extension_list
+    
+    def setEnvironmentSetting(self, environment_variable, value):
+        if VERBOSE:
+            print("SETTING {}={}".format(environment_variable, value))
+        self.settings[environment_variable] = value
+    
+    def setupEnvironment(self):
+        for k in self.settings:
+            if VERBOSE:
+                print("{}={}".format(k, str(self.settings[k])))
+            os.environ[k] = str(self.settings[k])
+    
+    def __str__(self):
+        out = ""
+        for k in list(self.settings.keys()):
+            out += "{}={}\n".format(k, self.settings[k])
+        
+        return out
+
+
+class TestFramework:
+    def __init__(self, in_files, test_comp, vt_install_dir):
+        self.test_comp = test_comp
+        
+        seperator = ':'
+        if sys.platform.startswith("win32"):
+            seperator = ';'
+        self.in_files = ""
+        for f in in_files:
+            self.in_files += "{}{}".format(os.path.join(os.getcwd(), f), seperator)
+        self.in_files = self.in_files[:-1]
+        
+        self.test_comp.test_environment.setEnvironmentSetting("VK_DEVSIM_FILENAME", self.in_files)
+        self.vt_install_dir = vt_install_dir
+    
+    def runAndCheck(self):
+        self.test_comp.run(self.vt_install_dir)
+        return self.test_comp.check()
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "vulkan_tools_install_dir",
+        help="Path to the install directory of the KhronosGroup/Vulkan-Tools repo."
+    )
+    
+    args = parser.parse_args()
+    
+    test_dict = {}
+    
+    vulkaninfo_path = args.vulkan_tools_install_dir
+    
+    if sys.platform.startswith("win32"):
+        vulkaninfo_path = os.path.join(vulkaninfo_path, "bin", "vulkaninfo.exe")
```

Comment:
I'm not a fan of OS-dependent code; but I don't have an easy way to make this sequence OS-independent, so I'm only grumbling about it.